--- /usr/share/openstack-puppet/modules/nova/manifests/compute/libvirt.pp   2014-06-26 11:03:34.000000000 -0400
+++ ./usr/share/openstack-puppet/modules/nova/manifests/compute/libvirt.pp  2014-07-17 21:44:12.456623055 -0400
@@ -109,7 +109,7 @@
     }
   }
 
-  if($::osfamily == 'RedHat' and $::operatingsystem != 'Fedora') {
+  if($::osfamily == 'RedHat' and $::operatingsystem != 'Fedora' and $::operatingsystemmajrelease < 7) {
     service { 'messagebus':
       ensure   => running,
       enable   => true,
--- /usr/share/openstack-puppet/modules/nova/manifests/params.pp    2014-06-26 11:03:34.000000000 -0400
+++ ./usr/share/openstack-puppet/modules/nova/manifests/params.pp   2014-07-17 22:00:38.827781399 -0400
@@ -41,11 +41,11 @@
       $root_helper                  = 'sudo nova-rootwrap'
       $lock_path                    = '/var/lib/nova/tmp'
       case $::operatingsystem {
-        'Fedora', 'RedHat': {
+        'Fedora', 'RedHat', 'CentOS': {
           $special_service_provider = undef
         }
-        'RedHat': {
-          if ($::operatingsystemrelease < 7) {
+        'RedHat', 'CentOS': {
+          if ($::operatingsystemmajrelease < 7) {
             $special_service_provider = 'init'
           } else {
             $special_service_provider = undef
--- /usr/share/openstack-puppet/modules/mysql/manifests/params.pp   2014-06-26 11:03:34.000000000 -0400
+++ ./usr/share/openstack-puppet/modules/mysql/manifests/params.pp  2014-07-17 22:05:42.630857393 -0400
@@ -37,8 +37,8 @@
             $provider = 'mysql'
           }
         }
-        'RedHat': {
-          if $::operatingsystemrelease >= 7 {
+        'RedHat', 'CentOS': {
+          if $::operatingsystemmajrelease >= 7 {
             $provider = 'mariadb'
           } else {
             $provider = 'mysql'
--- /usr/share/openstack-puppet/modules/apache/manifests/version.pp 2014-06-26 11:03:33.000000000 -0400
+++ ./usr/share/openstack-puppet/modules/apache/manifests/version.pp    2014-07-17 21:42:23.688227353 -0400
@@ -5,7 +5,7 @@
 class apache::version {
   case $::osfamily {
     'RedHat': {
-      if ($::operatingsystem == 'Fedora' and $::operatingsystemrelease >= 18) or ($::operatingsystem != 'Fedora' and $::operatingsystemrelease >= 7) {
+      if ($::operatingsystem == 'Fedora' and $::operatingsystemrelease >= 18) or ($::operatingsystem != 'Fedora' and $::operatingsystemmajrelease >= 7) {
         $default = 2.4
       } else {
         $default = 2.2
--- /usr/lib/python2.7/site-packages/packstack/puppet/templates/amqp.pp 2014-06-24 13:03:26.000000000 -0400
+++ ./usr/lib/python2.7/site-packages/packstack/puppet/templates/amqp.pp    2014-07-17 20:44:04.116453578 -0400
@@ -44,7 +44,7 @@
     }
 
     'RedHat', 'CentOS': {
-      if $::operatingsystemrelease >= 7 {
+      if $::operatingsystemmajrelease >= 7 {
         $config = '/etc/qpid/qpidd.conf'
       } else {
         $config = '/etc/qpidd.conf'
--- /usr/lib/python2.7/site-packages/packstack/puppet/templates/prescript.pp    2014-07-02 08:52:40.000000000 -0400
+++ ./usr/lib/python2.7/site-packages/packstack/puppet/templates/prescript.pp   2014-07-17 19:50:57.805970538 -0400
@@ -10,7 +10,7 @@
 }
 
 # For older RHEL-6 releases kernel/iptools does not support netns
-if $::operatingsystem in $el_releases and $::operatingsystemrelease < 7 {
+if $::operatingsystem in $el_releases and $::operatingsystemmajrelease < 7 {
   $info = "The RDO kernel that includes network namespace (netns) support has been installed on host $::ipaddress."
   $warning = " This is a community supplied kernel and is not officially supported by Red Hat. Installing this kernel on RHEL systems may impact your ability to get support from Red Hat."
 
--- /usr/lib/python2.7/site-packages/packstack/puppet/templates/mysql_install.pp    2014-06-24 13:03:26.000000000 -0400
+++ ./usr/lib/python2.7/site-packages/packstack/puppet/templates/mysql_install.pp   2014-07-17 20:42:59.352329584 -0400
@@ -1,6 +1,6 @@
 
 # on EL6 we need to wait for innodb changes before starting mysqld
-if $::operatingsystem in ['RedHat','CentOS','Scientific'] and $::operatingsystemrelease < 7 {
+if $::operatingsystem in ['RedHat','CentOS','Scientific'] and $::operatingsystemmajrelease < 7 {
     $manage_service = false
     service { 'mysqld':
       enable  => true,

